src/3d/materials/qgsabstractmaterialsettings.h

```diff
@@ -179,7 +180,7 @@ class _3D_EXPORT QgsAbstractMaterialSettings SIP_ABSTRACT
      * The \a technique argument specifies the rendering technique which will be used with the returned
      * material.
      */
-    virtual Qt3DRender::QMaterial *toMaterial( QgsMaterialSettingsRenderingTechnique technique, const QgsMaterialContext &context ) const = 0 SIP_FACTORY;
+    virtual Qt3DRender::QMaterial *toMaterial( const Qgs3DMapSettings &mapSettings, QgsMaterialSettingsRenderingTechnique technique, const QgsMaterialContext &context ) const = 0 SIP_FACTORY;
```

Comment:
@wonder-sk is this safe? Qgs3DMapSettings is a QObject, but AFAICS we are accessing this on non-main threads. (Both in master and more so with this PR). 

I'd argue we should be moving the properties required for material / symbol / entity creation to a different class which is cheap to copy and not a QObject, and then pass this new class around for uses like this.

Reply:
@nyalldawson Do you prefer we create a function in `Qgs3DMapSettings` to extract a simple struct (with the needed values) to pass as parameter in `toMaterial`?